qui/devices/backend.py

```diff
@@ -0,0 +1,277 @@
+# -*- encoding: utf8 -*-
+#
+# The Qubes OS Project, http://www.qubes-os.org
+#
+# Copyright (C) 2023 Marta Marczykowska-Górecka
+#                               <marmarta@invisiblethingslab.com>
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation; either version 2.1 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License along
+# with this program; if not, see <http://www.gnu.org/licenses/>.
+from typing import Set, Dict, Optional
+
+import qubesadmin
+import qubesadmin.exc
+import qubesadmin.devices
+import qubesadmin.vm
+
+import gi
+gi.require_version('Gtk', '3.0')  # isort:skip
+from gi.repository import Gtk, Gio  # isort:skip
+
+import gettext
+t = gettext.translation("desktop-linux-manager", fallback=True)
+_ = t.gettext
+
+
+class VM:
+    """
+    Wrapper for various VMs that can serve as backend/frontend
+    """
+    def __init__(self, vm: qubesadmin.vm.QubesVM):
+        self.__hash = hash(vm)
+        self._vm = vm
+        self.name = vm.name
+        self.vm_class = vm.klass
+
+    def __str__(self):
+        return self.name
+
+    def __eq__(self, other):
+        return str(self) == str(other)
+
+    def __lt__(self, other):
+        return str(self) < str(other)
+
+    def __hash__(self):
+        return self.__hash
+
+    @property
+    def icon_name(self):
+        """Name of the VM icon"""
+        try:
+            return getattr(self._vm, 'icon', self._vm.label.icon)
+        except qubesadmin.exc.QubesException:
+            return 'appvm-black'
+
+    @property
+    def is_dispvm_template(self) -> bool:
+        """
+        Is this VM a dispvm template?
+        """
+        return getattr(self._vm, 'template_for_dispvms', False)
+
+    @property
+    def is_attachable(self) -> bool:
+        """
+        Should this VM be listed as possible attachment target in the GUI?
+        """
+        return self.vm_class != 'AdminVM' and self._vm.is_running()
+
+    @property
+    def vm_object(self):
+        """
+        Get the qubesadmin.vm.QubesVM object.
+        """
+        return self._vm
+
+    @property
+    def should_be_cleaned_up(self):
+        """
+        VMs that should have the "shut me down when detaching device" option
+        """
+        return self._vm.auto_cleanup
```

Comment:
Not every VM has this attribute, wrap in `getattr()`

Reply:
I don't know why, but Github doesn't show me that I pushed a fix for that, but the code looks fine. So, done.